### PR TITLE
Avoid calculating bonds for water units when `ignoreHydrogens` is on

### PR DESCRIPTION
**Pull request by arose**
Avoid calculating bonds for water units when `ignoreHydrogens` is on

- Add `Water` trait to `Unit`
- Remove unsued code from `Structure.ofModel`
- Try reuse boundary in element-cross visual

### CODE REVIEW

File: src/mol-repr/structure/visual/bond-inter-unit-cylinder.ts [hunk 1]
```diff
@@ -246,7 +253,7 @@ export function InterUnitBondCylinderImpostorVisual(materialId: number): Complex
                 state.updateColor = true;
             }
 
-            if (newStructure.interUnitBonds !== currentStructure.interUnitBonds) {
+            if (hasStructureVisibleBonds(newStructure, newProps) && newStructure.interUnitBonds !== currentStructure.interUnitBonds) {
```
Comment (dsehnal) [inline src/mol-repr/structure/visual/bond-inter-unit-cylinder.ts:256]: Should the 1st condition be `hasStructureVisibleBonds(newStructure, newProps) !== hasStructureVisibleBonds(currentStructure, currentProps)`?

File: src/mol-repr/structure/visual/element-cross.ts [hunk 2]
```diff
@@ -68,10 +76,18 @@ export function createElementCross(ctx: VisualContext, unit: Unit, structure: St
         builder.add(s[0], s[1], s[2], e[0], e[1], e[2], i);
     }
 
+    const oldBoundingSphere = lines ? Sphere3D.clone(lines.boundingSphere) : undefined;
```
Comment (dsehnal) [inline src/mol-repr/structure/visual/element-cross.ts:79]: Move below the early return?

Comment (arose) [reply src/mol-repr/structure/visual/element-cross.ts]: done

File: src/mol-repr/structure/visual/bond-inter-unit-cylinder.ts (hunk 1 above)
Comment (arose) [reply src/mol-repr/structure/visual/bond-inter-unit-cylinder.ts:256]: no, not needed

**Pull request merged**